.github/workflows/rails_tests.yml

```diff
@@ -164,10 +168,22 @@ jobs:
           echo "postgres:*:*:postgres:postgres" > ~/.pgpass
           chmod 600 ~/.pgpass
 
-          bundle exec rake assets:precompile
+      - name: Cache precompiled assets
+        id: cache-assets
+        uses: actions/cache@v4
+        with:
+          path: public/assets
+          key: ${{ runner.os }}-assets-${{ hashFiles('**/app/assets/**/*', '**/app/javascript/**/*', '**/config/initializers/assets.rb') }}
+          restore-keys: |
+            ${{ runner.os }}-assets-
+
+      - name: Precompile assets
+        # if: steps.cache-assets.outputs.cache-hit != 'true'
+        run: bundle exec rails assets:precompile
 
       # Regenerate the graphql schema and fail if it has changed
       - name: 'Check HMIS GraphQL schema'
+        if: matrix.test_group == 'system_tests'
```

Comment:
Why does this step need to be tied to a test group? Would it run repeatedly if we didn't have this check?

Reply:
I could back this out. It's just that without the condition it runs for each element in the matrix which doesn't seem useful. It's not expensive though, we could leave it as is.

Reply:
It seems somewhat odd to group the system tests into this file if were just adding if statements for all the test sections to decide if it should run or not, maybe it should just be it's own file?

Reply:
We could make additional workflow files. I used conditional steps because it seemed more DRY, otherwise would copy/paste the entire workflow.

However there is something called a [reusable workflow](https://docs.github.com/en/actions/using-workflows/reusing-workflows). I haven't tried this before but it seems like it could work. We'd probably extract the common config from our current workflows. Maybe worth doing but I'd suggest we tackle it in another feature

Reply:
[tracking issue here](https://github.com/open-path/Green-River/issues/6409)